src/buttons.rs

```diff
@@ -11,86 +12,196 @@ const BUTTON_SPACING: f32 = 13.;
+            ButtonKind::Close => true,
+            ButtonKind::Maximize => supports_maximize,
+            ButtonKind::Minimize => supports_minimize,
+        };
+
+        let (buttons_left, buttons_right) =
+            Buttons::parse_button_layout(self.layout_config.clone())
+                .unwrap_or_else(Buttons::get_default_buttons_layout);
+
+        self.buttons_left = buttons_left.into_iter().filter(is_supported).collect();
+        self.buttons_right = buttons_right.into_iter().filter(is_supported).collect();
+    }
+
+    pub fn right_buttons_start_x(&self) -> Option<f32> {
+        match self.buttons_right.last() {
+            Some(button) => Some(button.x()),
+            None => None,
+        }
+    }
+
+    pub fn left_buttons_end_x(&self) -> Option<f32> {
+        match self.buttons_left.last() {
+            Some(button) => Some(button.end_x()),
+            None => None,
+        }
+    }
+
+    pub fn draw(
+        &self,
+        start_x: f32,
+        end_x: f32,
+        scale: f32,
+        colors: &ColorMap,
+        mouse_location: Location,
+        pixmap: &mut PixmapMut,
+        resizable: bool,
+        state: &WindowState,
+    ) {
+        let left_buttons_right_limit =
+            self.right_buttons_start_x().unwrap_or(end_x).min(end_x) - BUTTON_SPACING;
+        let buttons_left = self.buttons_left.iter().map(|x| (x, Side::Left));
+        let buttons_right = self.buttons_right.iter().map(|x| (x, Side::Right));
+
+        for (button, side) in buttons_left.chain(buttons_right) {
+            let is_visible = button.x() > start_x && button.end_x() < end_x
+                // If we have buttons from both sides and they overlap, prefer the right side
+                && (side == Side::Right || button.end_x() < left_buttons_right_limit);
+
+            if is_visible {
+                button.draw(scale, colors, mouse_location, pixmap, resizable, state);
+            }
         }
     }
 
-    pub fn iter(&self) -> std::array::IntoIter<Option<Button>, 3> {
-        [
-            Some(self.close.clone()),
-            self.maximize.clone(),
-            self.minimize.clone(),
-        ]
-        .into_iter()
+    fn parse_button_layout(sides: Option<(String, String)>) -> Option<ButtonLayout> {
+        if sides.is_none() {
+            return None;
+        }
+
+        let (left_side, right_side) = sides.unwrap();
+
+        let buttons_left = Buttons::parse_button_layout_side(left_side, Side::Left);
+        let buttons_right = Buttons::parse_button_layout_side(right_side, Side::Right);
+
+        if buttons_left.len() == 0 && buttons_right.len() == 0 {
+            warn!("No valid buttons found in configuration");
+            return None;
+        }
+
+        Some((buttons_left, buttons_right))
+    }
+
+    fn parse_button_layout_side(config: String, side: Side) -> Vec<Button> {
+        let mut buttons: Vec<Button> = vec![];
+
+        for button in config.split(",").take(3) {
```

Comment:
we can use char for single character splits:
```suggestion
        for button in config.split(',').take(3) {
```